woocommerce/woocommerce
# Add default values for Launch your store task

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/45114.


Set defaults for LYS task in the WooCommerce plugin update install/update routines when existing values do not exist.

#### Defaults when WooCommerce plugin is first installed. 

- Coming soon = `yes` for all sites.
- Launch status = `unlaunched`  for all sites.
- Store pages only = `yes` for existing sites,`no` for fresh sites.
- Private link = `yes` for all sites.
- Share key = 32 character URL friendly key

#### Defaults when WooCommerce plugin is updated

The same as above, except:

- Coming soon = `no` for all sites.
- Launch status = `launched`  for all sites.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Test when WooCommerce plugin is installed**

1. Create a fresh site without WooCommerce installed
2. Install `Code Snippets` plugin
3. Go to /wp-admin/plugin-editor.php?plugin=code-snippets%2Fcode-snippets.php&Submit=Select
4. Add the following code at the end of the file and save:

```php
function enable_lys_feature_flag( $features ) {
  return array_merge( $features, array( 'launch-your-store' ) );  
}

add_filter( 'woocommerce_admin_features', 'enable_lys_feature_flag', PHP_INT_MAX );
```

5.  Install WooCommerce using this branch https://betadownload.jetpack.me/data/woocommerce/add_lys-default-values/woocommerce-dev.zip

6. Confirm the following option values are set

- woocommerce_coming_soon: yes
- woocommerce_store_pages_only: no
- woocommerce_private_link: yes
- woocommerce_share_key:  32 character URL friendly key (alphanumeric)
- launch-status: `unlaunched`


**Test when WooCommerce is updated and the site is not fresh**

1. Create a fresh site and install WC
2. Install WC beta tester
3. Go to `/wp-admin/tools.php?page=woocommerce-admin-test-helper`
4. Enable `launch-your-store` feature flag
5. Run `wp option set fresh_site 0`
6. Run `wp option set woocommerce_version 8.7.0` to set `woocommerce_version` to a lower number.
7. Go to any WooCommerce Admin page and refresh.
8. Confirm the following option values are set

- woocommerce_coming_soon: no
- woocommerce_store_pages_only: yes
- woocommerce_private_link: yes
- woocommerce_share_key:  32 character URL friendly key (alphanumeric)
- launch-status: `launched`



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
